r/vignettes/developers/debugger.Rmd

```diff
@@ -0,0 +1,68 @@
+# Running R code with the C++ debugger
+
+As Arrow has C++ code at its core, debugging code can sometimes be tricky when 
+errors originate in the C++ rather than the R layer.  If you are adding new code
+which triggers a C++ bug (or find one in existing code), this can result in a 
+segfault.  If you are working in RStudio, the session is aborted, and you may 
+not be able to retrieve the error messaging needed to diagnose and/or report 
+the bug.  One way around this is to find the code that causes the error, and 
+run R with a C++ debugger.
+
+Firstly, load R with your debugger.  The most common debuggers are `gdb` and `lldb`
```

Comment:
You might want to note up here that if you're on macOS, you're probably going to be using lldb.

Reply:
Done